packages/medusa/src/services/analytics-config.ts

```diff
@@ -0,0 +1,101 @@
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import { AnalyticsConfigRepository as AnalyticsRepository } from "../repositories/analytics-config"
+import {
+  CreateAnalyticsConfig,
+  UpdateAnalyticsConfig,
+} from "../types/analytics-config"
+import { formatException } from "../utils"
+import UserService from "./user"
+
+type InjectedDependencies = {
+  analyticsConfigRepository: typeof AnalyticsRepository
+  manager: EntityManager
+}
+
+class AnalyticsConfigService extends TransactionBaseService {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly analyticsConfigRepository_: typeof AnalyticsRepository
+  protected readonly userService_: UserService
+
+  constructor({ analyticsConfigRepository, manager }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.analyticsConfigRepository_ = analyticsConfigRepository
+  }
+
+  async retrieve(userId: string) {
+    const manager = this.manager_
+
+    const analyticsRepo = manager.getCustomRepository(
+      this.analyticsConfigRepository_
+    )
+
+    const analyticsConfig = await analyticsRepo.findOne({
+      where: { user_id: userId },
+    })
+
+    if (!analyticsConfig) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `No analytics config found for user with id: ${userId}`
+      )
+    }
+
+    return analyticsConfig
+  }
+
+  async create(userId: string, data: CreateAnalyticsConfig) {
+    return await this.atomicPhase_(async (transactionManager) => {
```

Comment:
nit: atomic phase is not necessary here just use 

`const manager = this.transactionManager_ || this.manager_`